Blish HUD/Common/Gw2/Chat.cs

```diff
@@ -0,0 +1,120 @@
+                                            Keyboard.Stroke(VirtualKeyShort.KEY_V, true);
+                                            Thread.Sleep(50);
+                                            Keyboard.Release(VirtualKeyShort.LCONTROL, true);
+                                        }).ContinueWith(result => {
+                                            if (result.IsFaulted) {
+                                                Logger.Warn(result.Exception, "Failed to paste {text}", text);
+                                            } else if (prevClipboardContent != null)
+                                                ClipboardUtil.WindowsClipboardService.SetUnicodeBytesAsync(prevClipboardContent);
+                                        });
+                                });
+        }
+        /// <summary>
+        /// Returns the current string in the input field.
+        /// </summary>
+        public static async Task<string> GetInputText() {
+            if (!GameService.GameIntegration.Gw2IsRunning) return null;
+            byte[] prevClipboardContent = await ClipboardUtil.WindowsClipboardService.GetAsUnicodeBytesAsync();
+            await Task.Run(() => {
+                Focus();
+                Keyboard.Press(VirtualKeyShort.LCONTROL, true);
+                Keyboard.Stroke(VirtualKeyShort.KEY_A, true);
+                Keyboard.Stroke(VirtualKeyShort.KEY_C, true);
+                Thread.Sleep(50);
+                Keyboard.Release(VirtualKeyShort.LCONTROL, true);
+                Unfocus();
+            });
+            string inputText = await ClipboardUtil.WindowsClipboardService.GetTextAsync()
+                                                  .ContinueWith(result => {
+                                                       if (prevClipboardContent != null)
+                                                            ClipboardUtil.WindowsClipboardService.SetUnicodeBytesAsync(prevClipboardContent);
+                                                       return !result.IsFaulted ? result.Result : "";
+                                                   });
+            return inputText;
+        }
+        /// <summary>
+        /// Clears the input field.
+        /// </summary>
+        public static void Clear() {
+            if (!GameService.GameIntegration.Gw2IsRunning) return;
+            Task.Run(() => {
+                Focus();
+                Keyboard.Press(VirtualKeyShort.LCONTROL, true);
+                Keyboard.Stroke(VirtualKeyShort.KEY_A, true);
+                Thread.Sleep(50);
+                Keyboard.Release(VirtualKeyShort.LCONTROL, true);
+                Keyboard.Stroke(VirtualKeyShort.BACK);
+                Unfocus();
+            });
+        }
+        private static void Focus() {
+            Unfocus();
+            Keyboard.Stroke(VirtualKeyShort.RETURN);
+        }
+        private static void Unfocus() {
+            GameService.GameIntegration.FocusGw2();
```

Comment:
It GW2 actually needing focus at this point for it to work?  We used to need to set and unset focus primarily due to the way the old textbox implementation worked, but with that no longer in the way, the game should already have focus if the overlay is active.  

May be worth checking to see if the game is active (and not just running) so that we don't interrupt the user if they are doing something on another screen (not that we should be sending messages in chat on their behalf if they aren't actively in the game).

Reply:
I will try swapping the shortcuts with window messages like I did in Musician.
WM_PASTE, WM_COPY etc so I don't need to make these calls at all. I did try before. Probably have to make a change or two in PInvoke somewhere.
If that won't work I will check for the Activity of the window.